### PR TITLE
:seedling: Use official plantuml image for diagrams

### PR DESCRIPTION
**Pull request by killianmuldoon**
Use the official plantuml image from plantuml/plantuml https://hub.docker.com/r/plantuml/plantuml to build the images for the book and proposals.

This lets us get rid of a docker file with several pinned versions, and the Makefile in the docs folder.

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/area documentation

File: Makefile [hunk 1]
```diff
@@ -571,7 +574,18 @@ generate-metrics-config: $(ENVSUBST_BIN) ## Generate ./hack/observability/kube-s
 
 .PHONY: generate-diagrams
 generate-diagrams: ## Generate diagrams for *.plantuml files
-	$(MAKE) -C docs diagrams
+	$(MAKE) generate-diagrams-book
+	$(MAKE) generate-diagrams-proposals
+
+.PHONY: generate-diagrams-book
+generate-diagrams-book: ## Generate diagrams for *.plantuml files in book
+	-docker run -v $(ROOT_DIR)/$(DOCS_DIR):/$(DOCS_DIR)$(DOCKER_VOL_OPTS) plantuml/plantuml:$(PLANTUML_VER) /$(DOCS_DIR)/book/**/*.md
```
Comment (sbueringer) [inline Makefile:582]: What is the - prefix doing?

File: Makefile [hunk 2]
```diff
@@ -571,7 +574,18 @@ generate-metrics-config: $(ENVSUBST_BIN) ## Generate ./hack/observability/kube-s
 
 .PHONY: generate-diagrams
 generate-diagrams: ## Generate diagrams for *.plantuml files
-	$(MAKE) -C docs diagrams
+	$(MAKE) generate-diagrams-book
+	$(MAKE) generate-diagrams-proposals
+
+.PHONY: generate-diagrams-book
+generate-diagrams-book: ## Generate diagrams for *.plantuml files in book
```
Comment (sbueringer) [inline Makefile:581]: Do you know why we run plantuml on markdown files?

File: Makefile [hunk 3]
```diff
@@ -571,7 +574,18 @@ generate-metrics-config: $(ENVSUBST_BIN) ## Generate ./hack/observability/kube-s
 
 .PHONY: generate-diagrams
 generate-diagrams: ## Generate diagrams for *.plantuml files
```
Comment (sbueringer) [inline Makefile:576]: Not sure if it's a good idea, but wondering if we should run this in CI with a corresponding verify target
(I assume the resulting png files are reproducible)

**Comment by sbueringer** [comment]
Thx! Just a few questions

File: Makefile (hunk 1 above)
Comment (killianmuldoon) [reply Makefile:582]: Honestly don't know - didn't see it when I was moving over the targets. Removed now

File: Makefile (hunk 3 above)
Comment (killianmuldoon) [reply Makefile:576]: We're doing a basic verification - that the images don't disappear - using the link checker - I don't know if we really need something more advanced, but can look into it and take it as a follow up.

One issue is we also host a couple of svg images which are not generated by plantuml AFAIK

File: Makefile (hunk 1 above)
Comment (sbueringer) [reply Makefile:582]: Got lucky via google: https://stackoverflow.com/questions/3477292/what-do-and-do-as-prefixes-to-recipe-lines-in-make

> - means ignore the exit status of the command that is executed (normally, a non-zero exit status would stop that part of the build).

Sounds reasonable to drop it

File: Makefile (hunk 3 above)
Comment (sbueringer) [reply Makefile:576]: I leave it up to you. I'm fine with just this PR

File: Makefile (hunk 2 above)
Comment (killianmuldoon) [reply Makefile:581]: Appears to have been introduced here: https://github.com/kubernetes-sigs/cluster-api/pull/7033 but it looks like there no actual images generated from it so I've dropped them now.

File: Makefile (hunk 3 above)
Comment (killianmuldoon) [reply Makefile:576]: We have a bunch of images that are not generated from plantuml - list is below:

I'm not going to add a verify target in this PR - but I'll open an issue mentioning the current issues. I think this was supposed to be followed up from https://github.com/kubernetes-sigs/cluster-api/pull/7033/ but was missed.

```
        deleted:    docs/book/src/images/describe-cluster-disable-grouping.png
        deleted:    docs/book/src/images/describe-cluster-echo.png
        deleted:    docs/book/src/images/describe-cluster-how-grouping-works.png
        deleted:    docs/book/src/images/describe-cluster-show-conditions.png
        deleted:    docs/book/src/images/describe-cluster.png
        deleted:    docs/book/src/images/introduction.png
        deleted:    docs/book/src/images/runtime-sdk-lifecycle-hooks.png
        deleted:    docs/proposals/images/capi-provider-operator/fig3.png
        deleted:    docs/proposals/images/capi-provider-operator/fig4.png
        deleted:    docs/proposals/images/clusterctl-redesign/components.png
        deleted:    docs/proposals/images/conditions/cluster-provision-workflow.png
        deleted:    docs/proposals/images/conditions/upgrade-workflow.png
        deleted:    docs/proposals/images/externally-managed-cluster-infrastructure/infrastructure.png
        deleted:    docs/proposals/images/in-place-propagation/current-state.png
        deleted:    docs/proposals/images/in-place-propagation/optional-changes.png
        deleted:    docs/proposals/images/in-place-propagation/proposed-changes.png
        deleted:    docs/proposals/images/ipam-integration/consumption.png
        deleted:    docs/proposals/images/ipam-integration/sequence.png
        deleted:    docs/proposals/images/machine-states-preboot/Figure2.png
        deleted:    docs/proposals/images/runtime-hooks/runtime-hooks.png
        deleted:    docs/proposals/images/runtime-sdk/overview.png
        deleted:    docs/proposals/images/runtime-sdk/swagger-ui.png
```

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

File: Makefile (hunk 3 above)
Comment (killianmuldoon) [reply Makefile:576]: https://github.com/kubernetes-sigs/cluster-api/issues/9331

**Comment by killianmuldoon** [comment]
Cherry picks seem worthwhile to me to update very old dependency versions in supported branches.

**Pull request merged**